src/js/kits/model-kit-select/select.js

```diff
@@ -0,0 +1,29 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+
+const Select = (props) => {
+    const { fetchOptions, onChange } = props;
+    const [options, setOptions] = useState([{value: '', label: '(loading...)'}]);
+
+    useEffect(() => {
```

Comment:
Would recommend `useAsync` from `react-use` for this: https://github.com/streamich/react-use/blob/master/docs/useAsync.md.

Reply:
yeah, had a nice refactor into `AsyncSelect` using that.